src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -83,11 +83,11 @@ function initial_condition_convergence_test(x, t, equations::IdealGlmMhdEquation
     RealT = eltype(x)
     alpha = 0.25f0 * convert(RealT, pi)
     x_perp = x[1] * cos(alpha) + x[2] * sin(alpha)
-    B_perp = convert(RealT, 0.1) * sinpi(2 * x_perp)
+    B_perp = convert(RealT, 0.1) * sinpi(2 * (x_perp + t))
     rho = 1
     v1 = -B_perp * sin(alpha)
     v2 = B_perp * cos(alpha)
-    v3 = convert(RealT, 0.1) * cospi(2 * x_perp)
+    v3 = convert(RealT, 0.1) * cospi(2 * (x_perp + t))
```

Comment:
I think the reason why $t$ is not present here is that in the paper they write that 

![Screenshot 2025-06-29 101951](https://github.com/user-attachments/assets/1e67e7d8-7811-4b5f-b8ca-132e3346b8a8)

I guess we could make general `t` available here, but in that case I we should change this also in 1D

https://github.com/trixi-framework/Trixi.jl/blob/fb1c62552f34ab7ed407288b14c66e6b091d686a/src/equations/ideal_glm_mhd_1d.jl#L66-L80

Reply:
Can you double check that it converges properly for a non-integer value for the final time like `t_end=1.65`? In FLASH we rely on the fact that the solution loops back to the initial condition for integer time values (provided the domain is the correct size).

Reply:
> Can you double check that it converges properly for a non-integer value for the final time like `t_end=1.65`? In FLASH we rely on the fact that the solution loops back to the initial condition for integer time values (provided the domain is the correct size).

It does converge for general times. Otherwise, the time-dependent Dirichlet BC would not work.

Reply:
> I think the reason why t is not present here is that in the paper they write that...

Yes, I saw that the paper defines only the initial condition, but after some trial and error, I realized that the exact solution behaves like this. It works and provides convergence, and it's perfect to test the Dirichlet BC ;)

Reply:
Tests also pass :+1:

Reply:
> It was like this in 2D because there were a lot of tests with final time t e n d ∉ N . I assumed that it was going to be the same in 1D, but that was not the case 😅 

Oh, yeah that is annoying